### PR TITLE
PHP artisan error

### PR DESCRIPTION
**Pull request by kickb71**
If I run this command ("php artisan route:cache") it shows error message in the terminal

I have corrected and rectified all the files which is causing the error's not to run this command.

Issue no:#848 .

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -119,7 +118,7 @@
 
             Route::get('{module}', 'PermissionController@index')->name('permissions.module.index');
             Route::put('users/{id}', 'PermissionController@updateUserRoles')->name('permissions.module.update');
-            Route::put('roles/{id}', 'PermissionController@updateRolePermissions')->name('permissions.module.update');
+            Route::put('roles/{id}', 'PermissionController@updateRolePermissions')->name('permissions.module.role');
```
Comment (P4NK4J) [inline routes/web.php:121]: @kickb71 , I think the route name should specify what it's intending to. Can we rename this route to `role-update`, The name `role` seems quite generic. Let me know your thoughts.

**Review by P4NK4J — COMMENTED** [review]
Minor questions, let me know what you think.

File: routes/web.php (hunk 1 above)
Comment (kickb71) [reply routes/web.php:121]: That is what causing the error sir 
I have changed in the permisson controller to

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comments and make the necessary changes.

File: routes/web.php (hunk 1 above)
Comment (kickb71) [reply routes/web.php:121]: I will change the name to "permission.module.update.role" is that fine sir?

Comment (GauravGusain98) [reply routes/web.php]: You can use `permission.module.update-role`

Comment (GauravGusain98) [reply routes/web.php]: Also, does this route being used anywhere in the codebase? I can't see any changes related to the route name in the files from where this route is being called.

File: app/Http/Controllers/Settings/PermissionController.php [hunk 2]
```diff
@@ -47,7 +47,7 @@ public function updateUserRoles(UpdateUserRolesRequest $request)
 
     public function updateRolePermissions(UpdateRolePermissionsRequest $request)
     {
-        $this->authorize('settings.update', Permission::class);
+        $this->authorize('settings.update.role', Permission::class);
         $validatedData = $request->validated();
```
Comment (GauravGusain98) [inline app/Http/Controllers/Settings/PermissionController.php:51]: Revert these changes. They are not required. The old code is correct. 

Also, the name inside the authorize function is not the name of the route. It is the name of the function that is defined in the policy file. But for now, just revert these changes.

File: routes/web.php [hunk 3]
```diff
@@ -119,7 +118,7 @@
 
             Route::get('{module}', 'PermissionController@index')->name('permissions.module.index');
             Route::put('users/{id}', 'PermissionController@updateUserRoles')->name('permissions.module.update');
-            Route::put('roles/{id}', 'PermissionController@updateRolePermissions')->name('permissions.module.update');
+            Route::put('roles/{id}', 'PermissionController@updateRolePermissions')->name('permissions.module.update.role');
```
Comment (GauravGusain98) [inline routes/web.php:121]: I see that these 2 routes are not used anywhere in the codebase. Not sure why they were added. Do you know where these routes are used? 

Also, you can use the name `permissions.module.update-role` for the route

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Some minor changes required. Please make the changes and confirm if the routes are in use anywhere or not.

File: app/Http/Controllers/Settings/PermissionController.php (hunk 2 above)
Comment (kickb71) [reply app/Http/Controllers/Settings/PermissionController.php:51]: If we revert these changes we get error when we run the 'php artisan route:cache' command.

File: routes/web.php (hunk 3 above)
Comment (kickb71) [reply routes/web.php:121]: yes I can change the name to "permissions.module.update-role"

File: routes/web.php (hunk 1 above)
Comment (kickb71) [reply routes/web.php]: yes it is used in the "settings->permission->role.blade.php"

Comment (GauravGusain98) [reply routes/web.php]: Got it. I see that the URL is being used here. So for now, you can just change the name or route to `permission.module.update-role`.

File: app/Http/Controllers/Settings/PermissionController.php (hunk 2 above)
Comment (GauravGusain98) [reply app/Http/Controllers/Settings/PermissionController.php:51]: No, it will not throw an error. Just try it.

File: routes/web.php (hunk 3 above)
Comment (GauravGusain98) [reply routes/web.php:121]: yeah

File: app/Http/Controllers/Settings/PermissionController.php (hunk 2 above)
Comment (kickb71) [reply app/Http/Controllers/Settings/PermissionController.php:51]: I have tried it
error

Comment (GauravGusain98) [reply app/Http/Controllers/Settings/PermissionController.php]: I told you to just revert the changes of the PermissionController file. The route names were correct.

**Review by GauravGusain98 — APPROVED** [review]
Looks good to me.

**Comment by rathorevaibhav** [comment]
@GauravGusain98 do we need another review on this or it's good to merge? If good to merge, then please merge it after approval. For other PRs, also.

**Comment by GauravGusain98** [comment]
> @GauravGusain98 do we need another review on this or it's good to merge? If good to merge, then please merge it after approval. For other PRs, also.

Sure, I'll do this from now onwards.

**Pull request merged**